scripts/validate-android-test-env.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# This script validates that Android is set up correctly for the
+# testing environment.
+#
+# In particular, the config in ReactAndroid/build.gradle must match
+# the android sdk that is actually installed. Also, we must have the
+# right version of Java.
+
+# Check that Buck is working.
+if [ -z "$(which buck)" ]; then
+  echo "You need to install Buck."
+  echo "See https://buckbuild.com/setup/install.htm for instructions."
+  exit 1
+fi
+
+if [ -z "$(buck --version)" ]; then
+  echo "Your Buck install is broken."
+  if [ -d "/opt/facebook" ]; then
+    echo "FB laptops ship with a Buck config that is not compatible with open "
+    echo "source. FB Buck requires the environment to set a buck version, but "
+    echo "the open source version of Buck forbids that. Try setting:"
+    echo "export BUCKVERSION=ff27d5270ecaa92727cd5a19954e62298fa78f09"
```

Comment:
My concern is then if people know what they're doing and they do not want BUCKVERSION, they want to use their local buck setup, now all the scripts don't work.

I'm gonna bug the Buck folks to improve this but I don't expect an immediate fix.

How about this for now - I will adjust the instructions to say either you prefix your testing-running with setting the BUCKVERSION, or you can export it globally if you are willing to take that risk.